test/framework/clusterctl/client.go

```diff
@@ -99,7 +100,8 @@ func InitWithBinary(_ context.Context, binary string, input InitInput) {
 	_ = os.WriteFile(filepath.Join(input.LogFolder, "clusterctl-init.log"), out, 0644) //nolint:gosec // this is a log file to be shared via prow artifacts
 	var stdErr string
 	if err != nil {
-		if exitErr, ok := err.(*exec.ExitError); ok {
+		var exitErr *exec.ExitError
```

Comment:
```suggestion
		var exitErr exec.ExitError
```

Or is it not a pointer above on purpose?

Reply:
It's intentional.

In topology_plan exec.ExitError is an interface from k8s.io/util/exec
Here it is a struct from Go SDK exec

If I change it in either place I get the following warning: 
<img width="1342" alt="image" src="https://github.com/kubernetes-sigs/cluster-api/assets/4662360/ae3adbdf-4531-41fb-b759-8fc813ac1e53">

Reply:
But good (almost :)) catch!

Reply:
ah, struggled with that already the the first time until recognising it is k8s.io/.../exec

curious: couldn't we just use the same package in both cases? (I guess the answer is no)

Reply:
I honestly have no idea, because I don't know why we used the k/k exec package for topology dry run